### DISCUSSION TITLE
Automatically delete packages after 24h

### DISCUSSION THREAD

**Discussion by karstenBriksoft** [Homebrew/discussions#4508] [Everyday usage]
### Output of `brew config`

```shell
-na-
```

### Output of `brew doctor`

```shell
-na-
```

### Description of issue

I've noticed the following problem with Homebrew (and other package managers in general): after a while a ton of packages are installed and it's time to clean the package.
Of course it's possible to manually cleanup all packages every now and then, but why not try to avoid the problem in first place?

Why not install packages (that you already know will only be used once and then never again) on a temporary basis? What about a command like `temp-install` instead of `install`? I'd even go so far and make `install` a temporary thing and `perm-install` something that lasts.

Packages installed in a temporary fashion could automatically be removed upon next `cleanup`. Details of what a "good" installation period means probably varies from one day to one week or even a month.
Of course `cleanup` could also just ask the user to confirm deletion of packages before automatically removing them.

**Comment by SMillerDev** [comment]
I'm not sure what a temporary package would be, but why not run autocleanup in a cronjob?

**Comment by karstenBriksoft** [comment]
As far as I understand autocleanup would not remove packages that i manually installed. I'd first have to manually uninstall these packages.
I'd not call it a temporary package, I'd rather name it a temporary installation of a package: installing a package for a short amount of time compared to installing packages indefinitely until they are manually uninstalled.

**Reply by SMillerDev** [reply]
You could use `brew bundle` for that. It can reset your installed packages to a pre-defined list.
